File: src/dataset/cpp/luhn_exercism-Richard-m-j/gemini/GeminiFlash_luhn_v4.cpp
#include "luhn.h"
#include <string>

namespace luhn {
bool valid(std::string inputNumber) {
  int sum = 0;
  int length = 0;

  for (char c : inputNumber) {
    if (c != ' ') {
      if (!isdigit(c)) {
        return false;
      }
      length++;
    }
  }

  if (length <= 1) {
    return false;
  }

  bool doubleDigit = false;
  for (int i = inputNumber.length() - 1; i >= 0; --i) {
    if (inputNumber[i] != ' ') {
      int digit = inputNumber[i] - '0';
      if (doubleDigit) {
        digit *= 2;
        if (digit > 9) {
          digit -= 9;
        }
      }
      sum += digit;
      doubleDigit = !doubleDigit;
    }
  }

  return (sum % 10 == 0);
}
} // namespace luhn

File: src/dataset/typescript/luhn_exercism-wackerow/gemini/GeminiFlash_luhn_v4.ts
export default class Luhn {
  public static valid(_numbers: string): boolean {
    const numbers = _numbers.replace(/\s/g, '');

    if (numbers.length < 2 || !/^\d+$/.test(numbers)) {
      return false;
    }

    let sum = 0;
    for (let i = numbers.length - 1; i >= 0; i--) {
      let digit = parseInt(numbers[i], 10);
      if (isNaN(digit)) return false;

      if ((numbers.length - 1 - i) % 2 !== 0) {
        digit *= 2;
        if (digit > 9) {
          digit -= 9;
        }
      }
      sum += digit;
    }

    return sum % 10 === 0;
  }
}